diesel_tests/tests/combination.rs

```diff
@@ -26,7 +26,7 @@ fn union() {
     let data: Vec<_> = users
         .filter(id.le(tess.id))
         .union(users.filter(id.ge(tess.id)))
-        .positional_order_by(2) // name is the second column
+        .positional_order_by(OrderColumn::from(2)) // name is the second column
```

Comment:
The main drawback of the current implementation is the need to explicitly call `OrderColumn::from`.

On the other hand,  `1u32.desc()` didn't work either before, so explicit typing was required in some cases anyway.

We could add a helper method, e.g. something like `column_index(1)`. I need feedback on that

Reply:
As this is currently an internal API it's not worth to spend too much time on optimizing this, otherwise we could adjust the trait bounds on the `positional_order_by` methods with an additional `AsPositionalOrderExpr` that is implemented for:

* All types that implement `PositionalOrderExpr`
* `u32`

and that transforms `u32` to `OrderColumn`.

Reply:
I've taken a shot at that (except I named it IntoPositionalOrderExpr, as I'd need to return a new OrderColumn), can you tell me what you think?